gbus/abstractions.go

```diff
@@ -166,6 +166,9 @@ type Builder interface {
 	//ConfigureHealthCheck defines the default timeout in seconds for the db ping check
 	ConfigureHealthCheck(timeoutInSeconds time.Duration) Builder
 
+	//RetriesNum defines the number of retries upon error
+	RetriesNum(retries uint) Builder
```

Comment:
As there are going to be more configuration options that are currently hard coded but moving forward we would like to provide a way to configure I think we would probably be better off have one WithConfiguration(ConfigObject) that we can use to place on it additional config options moving forward.
WDYT ?